source/serverless-testing/README.rst

```diff
@@ -27,47 +27,42 @@ Serverless instance, so it is recommended to create one manually via the scripts
 in drivers-evergreen-tools that can be reused for the initial implementation of
 the tests before moving to Evergreen patches.
 
+.. _serverless directory in the drivers-evergreen-tools repository: https://github.com/mongodb-labs/drivers-evergreen-tools/tree/master/.evergreen/serverless
+
 Drivers MUST use the `create-instance.sh`_ script in ``drivers-evergreen-tools``
 to create a new Atlas Serverless instance for Evergreen tasks. The script writes
-two URIs to a YAML expansions file:
-
-- ``MULTI_ATLASPROXY_SERVERLESS_URI`` A URI to the load balancer fronting
-  multiple Atlas Proxy processes.
+a URI for the newly created instance to a YAML expansions file:
 
-- ``SINGLE_ATLASPROXY_SERVERLESS_URI`` A URI to one of the Atlas Proxy
-  processes.
+.. _create-instance.sh: https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/serverless/create-instance.sh
 
-The URIs can be read by drivers via the ``expansions.update`` Evergreen command.
-This will store the URIs into the ``SINGLE_ATLASPROXY_SERVERLESS_URI`` and
-``MULTI_ATLASPROXY_SERVERLESS_URI`` Evergreen expansions.
+- ``SERVERLESS_URI`` An SRV connection string to a load balancer fronting a
+  single Atlas Serverless proxy.
 
-.. _serverless directory in the drivers-evergreen-tools repository: https://github.com/mongodb-labs/drivers-evergreen-tools/tree/1ca6209825b6ed07ce90e24cda659143443709c8/.evergreen/serverless
-
-All tests MUST be run with wire protocol compression and authentication
-enabled.
+The ``expansions.update`` Evergreen command can be used to read this file and
+copy the URI into a ``SERVERLESS_URI`` environment variable.
 
 Test Runner Configuration
 =========================
 
-Drivers MUST add the username and password to both
-``SINGLE_ATLASPROXY_SERVERLESS_URI`` and ``MULTI_ATLASPROXY_SERVERLESS_URI`` to
-ensure that test clients can connect to the cluster. Drivers MUST use the final
-URI stored in ``SINGLE_ATLASPROXY_SERVERLESS_URI`` to configure internal clients
-for test runners (e.g. the internal MongoClient described by the `Unified Test
-Format spec <../../unified-test-format/unified-test-format.rst>`__).
+All tests MUST be run with wire protocol compression and authentication enabled.
+
+In contrast to the `Load Balancer testing`_, which has separate URIs for load
```

Comment:
This was probably discussed already, but why not have drivers spin up two separate instances per test run, one of which has only the single proxy behind the LB and one that has multiple?

Reply:
I did discuss this originally with Kevin and argued that the only reason we'd care about multiple mongoses from the driver's perspective is if we were directly connected to mongos(es) and needed to validate server pinning. If we're going through a load balancer, there is no benefit (to the driver) to have multiple mongoses behind the LB and it simply becomes a limitation for configuring fail points.

Testing a client with multiple mongoses is certainly of interest to the Atlas Serverless team, but I think it's reasonable for them to provide that test coverage themselves (and I expect they do).

----

> but why not have drivers spin up two separate instances per test run

Having Evergreen provision two different clusters for the same test run would also be a significant departure and likely complicate _every_ driver's test suite. At present, most of our Evergreen configs apart from the Atlas connectivity tests expect to feed the test suite a single connection URI (with optional variations for single/multi LB URIs). But in those cases, there's only a single cluster under test.

What you're describing is having the script provision two completely different clusters and feeding the test suite two entirely different connection strings.

That said, we _could_ introduce this new test environment (i.e. single proxy behind LB) as its own test environment, and require drivers to use the original Atlas Serverless project to provision a LB with multiple proxies; however, tests using the multi-proxy LB URI would never be able to test fail points reliably. This takes me back to the original point that there's no benefit to drivers in doing so.

Reply:
I understand there's be some benefit to running those tests against Atlas Serverless, although it does still seem like that'd be a matter of testing Atlas Serverless more than the driver itself. Unless I'm mistaken, I assume the driver's code paths are sufficiently tested by our non-serverless load balancer testing.

I don't mean to discount the importance of testing Atlas Serverless, since that's the reason we have connectivity tests in the first place, but I'm definitely of the opinion that it's not worth introducing the complexity (and CI overhead) at this time. Beyond the CI script changes, I think that would also require significant changes `runOnRequirement.server`, since we'd then need to differentiate between single and multi-proxy deployments.

If you're OK with taking no action here, feel free to resolve.